### ISSUE TITLE
GPU_NATIVE_TRAINING_SUBSETS entries missing 'optimizers' and 'losses' fields for 8 of 10 subsets

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#25]
## Problem

`GPU_NATIVE_TRAINING_SUBSETS` in `src/minicnn/cuda_native/capabilities.py` defines 10 training subset specs, but only the first two (`linear`, `flatten_linear`) include `'optimizers'` and `'losses'` keys:

```python
# linear — has both fields
{'name': 'linear', 'ops': [...], 'losses': [...], 'optimizers': ['SGD', 'Adam', 'AdamW', 'RMSprop'], ...},

# linear_relu_linear — missing both fields
{'name': 'linear_relu_linear', 'ops': [...], 'helper': '...', 'parity': '...'},

# conv_relu_pool_linear — missing both fields
{'name': 'conv_relu_pool_linear', 'ops': [...], 'helper': '...', 'parity': '...'},
```

The eight affected subsets are: `linear_relu_linear`, `flatten_linear_relu_linear`, `maxpool_linear`, `conv_linear`, `conv_relu_linear`, `conv_pool_linear`, `conv_relu_pool_linear`, `two_conv_relu_pool_linear`.

Any code reading `subset.get('optimizers', [])` or `subset['optimizers']` from these entries will get an empty list (or a `KeyError`), incorrectly implying no optimizers are supported — even though all eight support at minimum SGD. The top-level `'supported_optimizers': ['SGD', 'Adam', 'AdamW', 'RMSprop']` field does not distinguish by subset, so these entries are the authoritative per-subset capability source.

The `_validate_gpu_native_training_context` function in `_cuda_native_runtime.py` correctly blocks non-SGD optimizers for non-linear plans (which is the real enforcement gate), but the capabilities contract visible to introspection tools is incomplete.

## Fix

Add the missing fields to all eight entries, reflecting actual support:

```python
# two_linear_relu, pool_linear, and all conv subsets:
'losses': ['CrossEntropyLoss'],
'optimizers': ['SGD'],
```

For the `linear_relu_linear` and `flatten_linear_relu_linear` subsets, check whether Adam/AdamW/RMSprop support is implemented in `native_gpu_two_linear_relu_training_step` and list accordingly.

## Priority

P2 — incorrect (empty) optimizer/loss lists for 8 of 10 subsets make the capabilities matrix unreliable for introspection; `_validate_gpu_native_training_context` guards the runtime correctly, but the contract visible to callers is wrong.

**Comment by s9213712** [comment]
Closed by `e66919b` on `cuda-native-gpu-enablement`.

Resolution:
- `GPU_NATIVE_TRAINING_SUBSETS` now exposes `losses` and `optimizers` for every subset.
- Linear-only subsets keep full optimizer/loss support; all current composite/conv/pool subsets declare `CrossEntropyLoss` + `SGD`.
- Added regression coverage for this capability contract.
- Validated with the cuda_native GPU enablement subset: `163 passed, 4 skipped`.
